File: src/fibonacci/with_recursion/fibonacci.c
#include <stdio.h>

int fibonacci(int n);

int fibonacci(int n)
{
    if (n <= 1)
    {
        return(n);
    }
    else
    {
        /* if ( (n % 0xFF) == 0 )
        printf("%d ",n);
        */
        return( fibonacci(n-1) + fibonacci(n-2) );
    }
}

int main()
{
    int i;
    int limit = 8;

    for (i=0; i < limit; i++)
    {
        int fib = fibonacci(i);
        printf("fib(%d) = %d\n", i, fib);
    }

    return(0);
}

File: src/python/fibonacci.py
#!/usr/bin/env python

# fibonacci.py

# Global Variables
N=20

# Functions

# NOT LIKE in C
# int fib(int n) {...}
#
def fib(n):
    """Print a Fibonacci series up to n."""
    a,b = 0,1
    while a < n:
        print(a, end=' ')
        a,b = b,a+b
    print()
    return

def fib2(n):
    """Return a list containing the Fibonacci series up to n."""
    result = []
    a,b = 0,1
    while a<n:
        result.append(a)
        a,b = b,a+b
    return result

def main():
    fib(N)
    fib_N = fib2(N)
    print(fib_N)
    return

if __name__ == "__main__":
    # This is your program's main function
    main()

